RELEASE.md

```diff
@@ -1,23 +1,233 @@
+
+## Crates and versions
+
+This workspace consists of 4 crates:
+- `scylla`: main driver crate to be used by users. Depends on `scylla-cql` and `scylla-macros`.
+- `scylla-macros`: provides derive macros for users to be able to use their structs as UDTs / query parameters.
+- `scylla-cql`: low-level implementation of CQL binary protocol. Depends on `scylla-macros`.
+- `scylla-proxy`: mostly internal crate that aids testing. Depends on `scylla-cql`.
+
+We publish all 4 crates on [crates.io](https://crates.io/crates/scylla).
+
+Versions of the crates are decided with strict accordance to semver.
+For a sources on semver and semver in rust see:
+- https://doc.rust-lang.org/cargo/reference/resolver.html
+- https://doc.rust-lang.org/cargo/reference/semver.html
+- https://semver.org/
+- https://rust-lang.github.io/api-guidelines/necessities.html#public-dependencies-of-a-stable-crate-are-stable-c-stable (and some other parts of this book).
+
+What needs to be considered for releasing:
+- You can't change major version of scylla-cql / scylla-macros dependency in a minor update of scylla crate.
+For example, if scylla 0.13.0 depends on scylla-cql 0.4, then scylla 0.13.1 can't depend on scylla-cql 0.5.
+- We typically bump major versions of all 3 main crates together. I'm not sure if it's necessary - I remember there were
+arguments for that, but I can't remember them.
+
+
+## Documentation
+
+There are 2 places where documentation is published:
+- [docs.rs](https://docs.rs/scylla/latest/scylla/): API docs, generated automatically when publishing a crate to [crates.io.](https://crates.io/crates/scylla).
+- [Documentation book](https://rust-driver.docs.scylladb.com/stable/): generated and published by our CI (see [docs-pages.yaml](https://github.com/scylladb/scylla-rust-driver/blob/main/.github/workflows/docs-pages.yaml)) when there is a push to either `main` or some `branch-*` branch that updates something in `docs/` directory, or a push to some `v0.*` tag.
+
+Parameters relevant to releasing documentation book are configured in `docs/source/conf.py` and `docs/pyproject.toml` files.
+In `pyproject.toml` there is the `version` field - we do update it but I don't think it affects anything.
+In `docs/conf.py` there are `TAGS`, `BRANCHES`, `LATEST_VERSION`, `UNSTABLE_VERSIONS` and `DEPRECATED_VERSIONS` fields.
+
+`TAGS` and `BRANCHES` are lists of tags and branches for which documentation book will be built - those 2 lists add up to a list of versions you see when you click "expand" a version list in the left upper corner of [documentation book](https://rust-driver.docs.scylladb.com/stable/).
+
+`LATEST_VERSION` selects the element from the above list that will be the default ("stable") version shown when visiting https://rust-driver.docs.scylladb.com/stable/index.html.
+
+`UNSTABLE_VERSIONS` is a subset of `TAGS + BRANCHES`. If a version is on this list it will have a note in rendered documentation informing about instability:
```

Comment:
By `+` in `TAGS + BRANCHES` you mean set union?

Reply:
If so, then just for clarity let's write `\union` instead, or use the dedicated character: `∪`

Reply:
Those are Python lists - so I meant a concatenation of them. It should be identical to set union, unless someone decides to have a git branch and tag with a same name, which I hope we never do.